genie-web/src/main/java/com/netflix/genie/web/services/impl/LocalJobKillServiceImpl.java

```diff
@@ -176,6 +204,21 @@ public void onKillJobEvent(@NotNull final KillJobEvent event) throws GenieExcept
         this.killJob(event.getId(), event.getReason());
     }
 
+    @Override
+    public void registerCallbackForKill(
```

Comment:
it seems you're missing a deregister? Even if we don't have the outside service to shoulder tap you that it isn't connected to this node anymore I would think that after you invoke the callback on a kill signal you'd still somehow want to remove it from the map unless you're solely planning to rely on that other mechanism

Reply:
Makes sense. Done